tests/end_to_end_test/restore_test.go

```diff
@@ -270,19 +276,17 @@ func TestSnapshotRestore(t *testing.T) {
 		t.Fatalf("unexpected stat() results on output.zip directory %v %v", st, err)
 	}
 
-	restoreFailDir := t.TempDir()
-
 	// Attempt to restore snapshot with an already-existing target directory
 	// It should fail because the directory is not empty
-	_ = os.MkdirAll(restoreFailDir, 0o700)
-
 	e.RunAndExpectFailure(t, "snapshot", "restore", "--no-overwrite-directories", snapID, restoreDir)
 
 	// Attempt to restore snapshot with an already-existing target directory
 	// It should fail because target files already exist
-	_ = os.MkdirAll(restoreFailDir, 0o700)
-
```

Comment:
These seem spurious. Are these changes required? You are probably right that these directory creations are unnecessary, but it makes sense to put them in a separate PR to keep this self-contained.

Reply:
To me it makes sense to remove these now. I'm adding a new `--no-overwrite-` case, and my belief is it would be cleaner to have all three cases consistent: three similar test cases back-to-back that all follow the same pattern. The alternatives are (a) either the first two cases follow an unnecessary `mkdir` pattern that the new one doesn't, or (b) we add the new case with the unnecessary pattern for consistency among all three cases (only to be removed later).

If you feel strongly about it I can definitely move them to a separate PR, but IMO the changes aren't as spurious as they might seem.